### ISSUE TITLE
feat: store contract address in rlnCredentials.txt

### ISSUE THREAD

**Issue by richard-ramos** [status-im/nwaku#1238]
in the rlnCredentials.txt file we should also add the contract address that belongs to those credentials. Credentials are unique per contract, so it probably makes sense to store the contract address in the file, and that way we can avoid having to write the `--rln-relay-eth-contract` flag if it's already available in the credentials file

**Comment by rymnc** [comment]
Additionally, we should store the chainId as well

**Comment by rymnc** [comment]
Although, one could have the same credentials for different membership sets.
Currently, the rlnCredentials format is as follows -
```js
{
  "membershipKeyPair": {
    "idKey": uint8[],
    "idCommitment": uint8[]
  },
  "rlnIndex": uint256
}
```

We could have the schema extended in the following manner (if we consider that membership groups exist on only Ethereum smart contracts) -
```js
{
  "membershipKeyPair": {
    "idKey": uint8[],
    "idCommitment": uint8[]
  },
  "membershipGroups": {
    "chainId": {
      "contractAddress": string,
      "rlnIndex": uint256
    }
  }
}
```

We should also track these schemas in their own spec with versioning for backwards compatibility. 

wdyt? cc: @staheri14 @richard-ramos @jm-clius @s1fr0

**Comment by rymnc** [comment]
In favor of anonymity, we should not be using the same idCommitment across different membership groups. One membershipKeyPair should *only* be used in one membership group.

**Comment by staheri14** [comment]
> why not allowing users that stake double the membership fee to message at double rate

@s1fr0 That is a good suggestion (here is the tracking issue https://github.com/status-im/nwaku/issues/1264), but I think it is not captured by the initial solution, what you suggested is an n to 1 relationship between credentional and the group, whereas the initial solution covers the 1 to n relationship i.e., one credential and multiple groups. 

Nevertheless, supporting higher messaging rate with multiple valid credential requires some careful changes at the protocol level as well. We shall attempt those first before proceeding with any relevant update on the rln credetnial file.

**Comment by staheri14** [comment]
> As a side comment:membership fees are paid with an eth account so changing idcommitment but using same eth account to pay them would invalidate anonymity in same way.

Right, worth adding a security recommendation to the RFC about this.

**Comment by rymnc** [comment]
Yes, this would however, add complexity to the end user, and may result in a lot more user-prone errors, i.e choosing the wrong credential index. Furthermore, if we were to increase the messaging rate by having multiple membershipKeyPair's on one membershipGroup, the schema should be altered as follows -

```js
{
  "version": uint8
  "membershipGroups": {
    "chainId": {
      "contractAddress": {
        "rlnIndex": uint8,
        "membershipKeyPairs": MembershipKeyPair[]
      }
    }
  }
}
```

Which would allow us to enforce the relation that each membership group can have one or more membershipKeyPairs, but not the converse.
